### PR TITLE
[Fix #8985] Fix `Style/StringConcatenation` autocorrect generating invalid ruby

### LINKED ISSUE #8985: Style/StringConcatenation autocorrect doesn't properly escape double quotes when concatenated via single quotes
## Expected Behavior

When Rubocop encounters `'"' + "foo" + '"'` it should be autocorrected to `'"foo"'`

## Actual Behavior 
When Rubocop 1.1.0 encounters `'"' + "foo" + '"'` it currently autocorrects to `""foo""` thus leading to a syntax error:

```
test.rb:3:3: C: [Corrected] Style/StringConcatenation: Prefer string interpolation to string concatenation.
p '"' + "foo" + '"'
  ^^^^^^^^^^^^^^^^^
test.rb:3:5: E: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
p ""foo""
```

## Steps to reproduce the problem

Create a Ruby file with the following line and run `rubocop -A` in its directory
`'"' + "foo" + '"'` 

## RuboCop version

```
$ rubocop -V
1.1.0 (using Parser 2.7.2.0, rubocop-ast 1.1.0, running on ruby 2.7.2 x86_64-linux)
```
## Note

This is a bug specific to double quotes in single quotes. `"\"" + 'foo' + "\""` is properly autocorrected to `'"foo"'`.

### PR DESCRIPTION
**Pull request by tejasbubane**
When single quoted strings

Closes #8985 

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/string_concatenation_spec.rb [hunk 1]
```diff
@@ -158,4 +158,26 @@
...
+        "'" + "foo" + '"' + "bar"
+        ^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
+      RUBY
+
+      expect_correction(<<-RUBY)
+        "'foo'"
+        "'foo'"
+        "'foo'"
+        "'foo'bar"
+      RUBY
```
Comment (dvandersluis) [inline spec/rubocop/cop/style/string_concatenation_spec.rb:180]: This doesn't look right to me, it's changing the actual string value now. The original examples returned these values:

```
"foo"
"foo'
'foo"
'foo"bar
```

Comment (tejasbubane) [reply spec/rubocop/cop/style/string_concatenation_spec.rb]: @dvandersluis You are right. I have fixed this now.

File: lib/rubocop/cop/style/string_concatenation.rb [hunk 2]
```diff
@@ -106,7 +106,13 @@ def replacement(parts)
               end
             end
 
-          "\"#{interpolated_parts.join}\""
+          "\"#{handle_empty_quotes(interpolated_parts).join}\""
+        end
+
+        def handle_empty_quotes(parts)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/string_concatenation.rb:112]: What do you mean by "empty quotes" here? Is this an empty string that's basically just ""?

Comment (tejasbubane) [reply lib/rubocop/cop/style/string_concatenation.rb]: Renamed to `handle_quotes`. I mean handle when parts contain quote only like `'"'` or `"'"`.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**